internal/integration_test/engine/adhoc_test.go

```diff
@@ -558,50 +558,53 @@ func testCloseInFlight(t *testing.T, r wazero.Runtime) {
 
 func testMemOps(t *testing.T, r wazero.Runtime) {
```

Comment:
so this was exactly the same pattern in trivy and karmen - calling exported functions which _might_ grow memory. But in order to catch this bug, I had to reuse the same api.Function, so I fixed it in that way.

Reply:
and verified that without the patch, this test fails!